Hai Reveal conversations: bug fix to log entries, remove looping dialog and action choices from part 6 (#7707)
endless-sky/endless-sky: data/hai/hai reveal 6 end.txt

			`	Apparently, since the Syndicate is technically part of the Republic, they are awarded no special diplomatic status and have been busy with corporate matters elsewhere. Xilin finds you between proceedings and warns you, "Terry seems to have disappeared from Syndicate communications, though I won't hazard a guess as to their internal politics." A warning that carries no surprise at this point.`
			`	It seems this station has seen serious renovations in the past weeks. You find yourself standing with the diplomats in a room now called the grand Borealis Chamber, "Where Mountains Touch The Sky," as Teeneep makes the first publicly broadcast speech from the Hai to humanity. The three heads of state announce a resolution granting safe passage to all Hai in human space, as long as they register with customs on Mountaintop before entering, and Teeneep expresses similar sentiments for humans visiting Hai space. It is announced that extensive surveillance fleets will be posted on both sides of the wormhole to ensure safety. This meeting will now turn into a multi-day summit to discuss the creation of a formal treaty.`

> 			`	As you leave the summit for lunch, Xilin and Sayari intercept you. "Trouble. We need to talk," says Xilin.`

Reviewer: ```suggestion
			`	As you leave the summit for lunch, Xilin and Sayari intercept you.`
			`	"Trouble. We need to talk," says Xilin. You follow them to a private meeting room.`
```
Looks better, still could be just a bit more improved. I don't like telling the player what they feel, better for other characters to express that. I'd like to give the player an option to feel frustrated, though, and let Sayari or Xilin talk about "how easy it would be for it to be over".

Author: > You follow them to a private meeting room.

I'm trying not to lengthen an already quite long scene by adding another paragraph. It doesn't say where you are when the conversation occurs, so it is reasonable to assume you're in a place where you can discuss such matters. Sayari has been a diplomat longer than a human lifespan, so she'd have the experience to know when they can safely talk.

Reviewer: I'd imagine you're leaving the Borealis Chamber the moment Sayari and Xilin meet you again, then they would bring you down to the meeting rooms. Otherwise, the player might feel like they're talking out in the open. Only when you get Sayari to roll her eyes, you go to the meeting rooms.

Author: Okay. I put it back in.

			choice
				`	"What's wrong?"`
					goto need2talk
				`	"Hey, don't worry. Whatever it is, I'm sure we've been through worse."`

> 			`	Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted." You follow her to a private meeting room.`

Reviewer: ```suggestion
			`	Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted."`
```

		log `Have learned that during the crisis a single ship loaded with kidnapped Hai was misplaced and abandoned. They require rescue and may indicate the presence of a loose end.`
		conversation
			`Arriving on Mountaintop brings with it full diplomatic regalia. The Chancellor of the Free Worlds and President of the Republic are in their most respectable suits, Alondo and Xilin have the traditional symbols of office (though Alondo's looks like the Free Worlds have perhaps not finalized their design just yet), and dress uniforms laden with medals abound on both sides.`

> 			`	Apparently, the Syndicate is technically part of the Republic and awarded no special diplomatic status, so they've been busy with corporate matters elsewhere. Xilin finds you between proceedings and warns you, "Terry seems to have disappeared from Syndicate communications, though I won't hazard a guess as to their internal politics." A warning that carries no surprise at this point.`

Reviewer: ```suggestion
			`	Apparently, since the Syndicate is technically part of the Republic, they are awarded no special diplomatic status and have been busy with corporate matters elsewhere. Xilin finds you between proceedings and warns you, "Terry seems to have disappeared from Syndicate communications, though I won't hazard a guess as to their internal politics." A warning that carries no surprise at this point.`
```
Maybe this could be made better than my suggestion.

			`	It seems this station has seen serious renovations in the past weeks. You find yourself standing with the diplomats in a room now called the grand Borealis Chamber, "Where Mountains Touch The Sky," as Teeneep makes the first publicly broadcast speech from the Hai to humanity. The three heads of state announce a resolution granting safe passage to all Hai in human space, as long as they register with customs on Mountaintop before entering, and Teeneep expresses similar sentiments for humans visiting Hai space. It is announced that extensive surveillance fleets will be posted on both sides of the wormhole to ensure safety. This meeting will now turn into a multi-day summit to discuss the creation of a formal treaty.`
			`	As you leave the summit for lunch, Xilin and Sayari intercept you and seem quite worried. `
			choice

> 				`	"What's wrong?"`
> 					goto need2talk
> 				`	"Hey, don't worry. Whatever it is, I'm sure we've been through worse."`
> 			`	Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted." You follow her to a private meeting room.`
> 				goto whut
> 			label need2talk
> 			`	"We need to talk," says Xilin`
> 			`	You feel the deepest of sighs settle into your bones, but at this point you just follow them to a private meeting room. It would have been too easy if it was all over, wouldn't it?`
> 			label whut

Reviewer: ```suggestion
				`	"What's wrong?"`
					goto need2talk
				`	"Hey, don't worry. Whatever it is, I'm sure we've been through worse."`
			`	Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted." You follow her to a private meeting room.`
				goto whut
			label need2talk
			`	"We need to talk," says Xilin.`
			`	You feel the deepest of sighs settle into your bones, but at this point you just follow them to a private meeting room. It would have been too easy if it was all over, wouldn't it?`
			label whut
```
This could be rearranged, instead of telling the player what they feel. "We need to talk" could come first, then have the player ask what's wrong.

Author: I changed it to "Trouble. We need to talk,"

> 				`	"What's wrong?"`
> 					goto need2talk
> 				`	"Hey, don't worry. Whatever it is, I'm sure we've been through worse."`
> 			`	Sayari rolls her eyes and says, "Don't relax yet, Captain <last>. There is one more minor disaster to be averted."`
> 				goto whut
> 			label need2talk
> 			`	Sayari says, "It would have been too easy if it was all over. We have one more minor disaster to avert first."`
> 			label whut
> 			`	Xilin begins. "It seems that somewhere along the way, a shipload of kidnapped Hai didn't make it to a processing area. The Hai were left on Sunracer, and the ship appears to have disappeared."`

Reviewer: ```suggestion
				`	"What's wrong?"`
					goto whut
				`	"Really? I thought we had figured everything out."`
				goto need2talk
				`	"Hey, don't worry. Whatever it is, I'm sure we've been through worse."`
			`	Sayari rolls her eyes and says, "Don't relax yet, Captain <last>. There is one more minor disaster to be averted."`
				goto whut
			label need2talk
			`	Sayari says, "It would have been too easy if it was all over. We have one more minor disaster to avert first."`
			label whut
			`	Xilin begins. "It seems that somewhere along the way, a shipload of kidnapped Hai didn't make it to a processing area. The Hai were left on Sunracer, and the ship appears to have disappeared."`
```
While I recommended moving that line to Sayari, it's out of place with what the player says to get to that dialogue, and works better this way.

Author: Your changes were missing a tab. I got them working, though.

> 			`	Xilin begins. "It seems that somewhere along the way, a shipload of kidnapped Hai didn't make it to a processing area. The Hai were left on Sunracer, and the ship appears to have disappeared."`

Reviewer: I still think it should go like this: 
>As you leave the summit for lunch, Xilin and Sayari intercept you and seem quite worried. 
>"Trouble. We need to talk," says Xilin.

>"What's wrong?" goto whut
>"Hey, don't worry. Whatever it is, I'm sure we've been through worse."

>Sayari rolls her eyes and says, "Maybe so, but there is one more minor disaster to be averted." You follow her to a private meeting room.

>label whut
>Xilin begins. "It seems that somewhere along the way, a shipload of kidnapped Hai didn't make it to a processing area. The Hai were left on Sunracer, and the ship appears to have disappeared." `

Author: In the "whut" branch, Xilin needs to take you to a private meeting room.

Reviewer: Yes, I just realised that.

Author: I rearranged the text and omitted the part about where to talk. Does it work now?

		log `Have learned that during the crisis a single ship loaded with kidnapped Hai was misplaced and abandoned. They require rescue and may indicate the presence of a loose end.`
		conversation
			`Arriving on Mountaintop brings with it full diplomatic regalia. The Chancellor of the Free Worlds and President of the Republic are in their most respectable suits, Alondo and Xilin have the traditional symbols of office (though Alondo's looks like the Free Worlds have perhaps not finalized their design just yet), and dress uniforms laden with medals abound on both sides.`

> 			`	Apparently, since the Syndicate is technically part of the Republic, they are awarded no special diplomatic status and have been busy with corporate matters elsewhere. Xilin finds you between proceedings and warns you, "Terry seems to have disappeared from Syndicate communications, though I won't hazard a guess as to their internal politics." A warning that carries no surprise at this point.`

Reviewer 2: I'm not sure I like the entirety of this first line swap. It feels like Xilin is acting like he knows things, when if he's warning you it'd at best be a "don't go stick your foot in it over there" type deal.

Author: I rearranged it again. How does it look now?

Reviewer 2: Yes I like that

			choice
				`	"The Hai are no longer a secret in human space, but they are concerned how you ended up here."`
					goto revealed
					to display
						not "Hai Reveal [C02] Cleanup: label revealed"
				`	"What happened to the ship that brought you here?"`
					goto kidnappers
					to display
						not "Hai Reveal [C02] Cleanup: label kidnappers"
				`	"Well, lets get you home before anything else happens."`
					goto depart
			label revealed
			action
				set "Hai Reveal [C02] Cleanup: label revealed"
			`	"Ah, we thought we saw something about that, but our comms are spotty, so we didn't rely on it. We all actually signed up to go to Mountaintop to take part in the facade, so it seemed especially suspicious."`
			`	He shrugs. "An abundance of caution seemed warranted; we were kidnapped, after all."`
				goto responses
			label kidnappers
			action
				set "Hai Reveal [C02] Cleanup: label kidnappers"
			`	"We were en route to somewhere after the ship left Hai space. We don't really know what happened, but they received some kind of message that left them very concerned. They said something about hiding somewhere safe and landed here."`

> 			`	At this point, he frowns in the way Hai do. "We were set down here for days and days. Our kidnappers couldn't seem to work out what they should do. Eventually, a woman landed next to us in this ship here," he explains, gesturing at the Star Barge. "She had them ferry us and all our stuff out to this ship, and then she took charge and flew off with them. That was when we sent the message. We were partly afraid it was some kind of elaborate trap when we saw that we could communicate to a Hai address directly. We never used to be able to communicate between human space and ours like that. I guess we were lucky this time."`

Reviewer: How do the Hai frown in the way they do?

Author: I just removed the redundant information that the Hai is frowning in the way that Hai frown, since we know Hai always frown in the way that Hai frown.

			choice
				`	"The Hai are no longer a secret in human space, but they are concerned how you ended up here."`
					goto revealed
					to display
						not "Hai Reveal [C02] Cleanup: label revealed"
				`	"What happened to the ship that brought you here?"`
					goto kidnappers
					to display
						not "Hai Reveal [C02] Cleanup: label kidnappers"

> 				`	"Well, lets get you home before anything else happens."`

Reviewer: Purely syntax changes. 
```suggestion
				`	"Well, let's get you home before anything else happens."`
```

					goto depart
			label revealed
			action
				set "Hai Reveal [C02] Cleanup: label revealed"
			`	"Ah, we thought we saw something about that, but our comms are spotty, so we didn't rely on it. We all actually signed up to go to Mountaintop to take part in the facade, so it seemed especially suspicious."`
			`	He shrugs. "An abundance of caution seemed warranted; we were kidnapped, after all."`
				goto responses
			label kidnappers
			action
				set "Hai Reveal [C02] Cleanup: label kidnappers"
			`	"We were en route to somewhere after the ship left Hai space. We don't really know what happened, but they received some kind of message that left them very concerned. They said something about hiding somewhere safe and landed here."`

> 			`	At this point, he frowns. "We were set down here for days and days. Our kidnappers couldn't seem to work out what they should do. Eventually, a woman landed next to us in this ship here," he explains, gesturing at the Star Barge. "She had them ferry us and all our stuff out to this ship, and then she took charge and flew off with them. That was when we sent the message. We were partly afraid it was some kind of elaborate trap when we saw that we could communicate to a Hai address directly. We never used to be able to communicate between human space and ours like that. I guess we were lucky this time."`

Reviewer 2: Remember how earlier we described how Hai naturally are somewhat more alien than you might expect from encountering those familiar with humans?
This was meant to be a subtle nod to the possibility that you've seen enough of them to interpret them sometimes rather than have them adapt to you all the time.

Reviewer: Perhaps, but I felt it was not very meaningful unless it was described to the player. I don't know how a Hai frowns, anyway.

Author: For it to be meaningful, we would have to describe some aspects of the Hai facial features, but the game is very careful not to describe the various alien species in detail. Rather, we leave it to the player's imagination.

Reviewer 2: Which is why we describe such things like familiarity with those foreign details in abstract terms.
I'm not overly hung up on this point though, just felt I should clarify that there was intention to it.
If people don't like it it can stay removed.

Author: I just made a change. How about this? "At this point, he twists his face in a non-human manner you recognize as a Hai frown."

	to offer
		has "Hai Reveal [C02] Cleanup: active"
	on enter Mirfak

> 		dialog "A message from Syndicate Corporate Headquarters pops up before you can depart Mirfak. 'Captain <last>, one of our executives, Terry Adrianopoulos, has gone missing; if you see her, let her know she will lose her position if she doesn't return to her duties soon.'"

Reviewer 2: Yep, this is an outdated line, but I think it should actually come from someone you know, either Moray or Remington, leaning towards Moray.

Author: I used Moray since he shows up less, and thus is less likely to cause complications in future changes to the story.

		conversation

> 			`	Your arrival is unobstructed as you land on this mild, habitable world that appears to be in an advanced, but retreating, ice age. No one comes to greet you at the spaceport, either. As you wander into the single major town, built on a promontory overlooking a wide river mouth, you notice that the people here seem to be happy and the town appears to be remarkably modern. This place has been looked after by someone who knew how to source the necessary import goods to support it on this sparse world of grasses and something resembling pine trees - someone who was protege to the CEO of the Syndicate, for example...`

Reviewer 2: Updating for consistency.
```suggestion
			`	Your arrival is unobstructed as you land on this mild, habitable world that appears to be in an advanced, but retreating, ice age. No one comes to greet you at the spaceport, either. As you wander into the single major town, built on a promontory overlooking a wide river mouth, you notice that the people here seem to be happy and the town appears to be remarkably modern. This place has been looked after by someone who knew how to source the necessary import goods to support it on this sparse world of grasses and something resembling pine trees - someone who was Chief of Operations of the Syndicate, for example...`
```

			`	She gives you a hard stare, but her resolve isn't up to the task for more than a couple seconds. She looks away, and you can see her blinking away more tears. "Is this to be my end, then?" she asks. "Is this the price I pay for my efforts?"`
			label decide
			choice
				`	"Am I to be judge, jury, and executioner?"`
					goto judge
					to display
						not "Hai Reveal [C05] Cleanup: label judge"
				`	(Spare her.) "The whole galaxy hunts you now. Let that be your justice."`
					goto spare
				`	(End her.) "The price you pay is justice."`
					goto end

> 				`	(Spare her.) "I've sinned more than you. If people like us decided what is right or wrong, the galaxy would be cinders."`

Reviewer: ```suggestion
				`	(Spare her.) "I've sinned more than you. If people like us decided what is right or wrong, the galaxy would be in cinders."`
```

			`	She gives you a hard stare, but her resolve isn't up to the task for more than a couple seconds. She looks away, and you can see her blinking away more tears. "Is this to be my end, then?" she asks. "Is this the price I pay for my efforts?"`
			label decide
			choice

> 				`	"Am I to be judge, jury, and executioner?"`
> 					goto judge
> 					to display

Reviewer 2: I would suggest putting these back in because I think the scene is potentially most powerful with no words said. I think the players should have that option.
```suggestion
				`	"Am I to be judge, jury, and executioner?"`
					goto judge
				`	(Spare her silently.)`
					goto spare
				`	(End her wordlessly.)`
					goto end
					to display
```

Author: I restored both of the silent options and updated the `to display` logic, which your suggested change broke. There are *way* too many options, so I changed which ones occur before and after asking if you're the judicial system.

Before asking if you're the judicial system, you get the silent options and the "I don't feel like thinking about it" options:

1. Am I to be judge, jury, and executioner?
2. Silent responses
3. This is for all the trouble you've caused me, personally.
4. Movie review responses

After asking if you're the judicial system, you lose the movie review options and gain the "I thought about it" options:

1. Silent responses
2. There is no other home for you now. Let that be your justice.
3. Justice is the price you pay.
4. I've sinned more than you... wrong... galaxy... cinders.
5. This is for all the trouble you've caused me, personally.

Reviewer 2: I like it.

						not "Hai Reveal [C05] Cleanup: label judge"

> 				`	(Spare her.) "The whole galaxy hunts you now. Let that be your justice."`

Reviewer 2: Since the whole point is that she's not actually hunted, you're the only person who knows so long as she stays hidden, it's not really appropriate to say she's hunted.
```suggestion
				`	(Spare her.) "There is no other home for you now. Let that be your justice."`
```

					goto spare

> 				`	(End her.) "The price you pay is justice."`

Reviewer 2: ```suggestion
				`	(End her.) "Justice is the price you pay."`
```

					goto end
				`	(Spare her.) "I've sinned more than you. If people like us decided what is right or wrong, the galaxy would be in cinders."`
					goto spare

> 				`	(End her.) "This is for all the trouble you put me through."`

Reviewer 2: I suggest this change because it emphasises that the first two spoken choices are moralising options, whereas the second two are personal options.

I consider this an optional suggestion though.
```suggestion
				`	(End her.) "This is for all the trouble you've caused me, personally."`
```

					goto end
				`	(Spare her.) "Consider this my movie review."`
					goto spare
					to display
						has "Hai Reveal [C05] Cleanup: label judge"
				`	(End her.) "Consider this my movie review."`
					goto end
					to display
						has "Hai Reveal [C05] Cleanup: label judge"
			label spare

> 			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself though, you think maybe the future is worth the sacrifice made in the attempt.`

Reviewer: ```suggestion
			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself, though, you think: "Maybe the future is worth the sacrifice made in the attempt."`
```
I can't tell what this means. The future is worth the sacrifice?

> 			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself, though, you think: "Maybe the future is worth the sacrifice made in the attempt."`

Reviewer 2: Can I summon @EjoThims to this specific line? I'm thinking maybe to avoid the clunky punctuation we should go more like:
```suggestion
			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself though, you wonder if perhaps the future is worth the sacrifice made in the attempt.`
```

Reviewer: Yup. This was the best I could do with this line, but what does it mean?

Reviewer 3: @UnorderedSigh The most recent suggestion is much better punctuation, but it is still a confusing wording. Especially in reference to which sacrifice and attempt.

I get what it’s referring to, just not what it’s trying to say about what it’s referring to.

Author: The sentence confuses me too, so I've been wary of touching it.

Author: > The most recent suggestion is much better punctuation

I already applied the most recent suggestion.

> 			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself though, you wonder if perhaps the future is worth the sacrifice made in the attempt.`

Author: Yes. The particular morality is already in the `choice`, so it makes sense to remove it here.

					goto spare
				`	(End her wordlessly.)`
					goto end

				# Justifications are only available after asking for clarification:
				`	(Spare her.) "There is no other home for you now. Let that be your justice."`
					goto spare
					to display
						has "Hai Reveal [C05] Cleanup: label judge"
				`	(End her.) "Justice is the price you pay."`
					goto end
					to display
						has "Hai Reveal [C05] Cleanup: label judge"
				`	(Spare her.) "I've sinned more than you. If people like us decided what is right or wrong, the galaxy would be in cinders."`
					goto spare
					to display
						has "Hai Reveal [C05] Cleanup: label judge"

				# The callous option is available regardless, but it's after the "I thought about it" options:
				`	(End her.) "This is for all the trouble you've caused me."`
					goto end

				# Joke options are only available if you don't care enough to ask for clarification:
				`	(Spare her.) "Consider this my movie review."`
					goto spare
					to display
						not "Hai Reveal [C05] Cleanup: label judge"
				`	(End her.) "Consider this my movie review."`
					goto end
					to display
						not "Hai Reveal [C05] Cleanup: label judge"

			label spare

> 			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself though, you cannot bring yourself to be her executioner.`

Reviewer: ```suggestion
			`	You draw your weapon and point it at her, your thumb on the safety. When you look inside yourself, though, you cannot bring yourself to be her executioner.`
```

				log "People" "Terry Adrianopoulos" `During the Hai-human crisis, Terry was arrested when it seemed like war might break out, but was released when the threat of conflict ended. She immediately vanished and was later identified in association with some lingering kidnappers. It is unclear what happened to them, but it is implied that she dealt with them in some way. It turned out that she had been a lone operative infiltrating the Syndicate for many years, with the goal of seizing control and rebuilding the government to be less corrupt, but she condoned kidnapping and exploitation to achieve it. Her story ended on the bluffs of Aspiration, though no one may ever know it.`
			`	The retort of the weapon is taken away on the wind.`
				goto return
			label judge
			action
				set "Hai Reveal [C05] Cleanup: label judge"
			`	"You'd better be," she says. "You and I both know that this can't become public. If you take me to the Syndicate they will do away with me if they find out, and I doubt they could manage to not."`

> 			`	"Oh, Alastair did tell me to say you're fired if you don't get back quickly... that was over a week ago now," you tell her.`

Reviewer: Alastair no longer says this.

Author: Fixed.
